File: src/java/DisariumNumber.java
package java;

public class DisariumNumber {

    private static int countDigits(long num) {
        int len = 0;

        while(num > 0) {
            len++;
            num /= 10;
        }

        return len;
    }

    public static boolean isDisariumNumber(long num) {
        long tempNum = num;
        long sum = 0;
        int digits = countDigits(num);

        while(tempNum > 0) {
            sum += Math.pow(tempNum%10, digits--);
            tempNum /= 10;
        }

        return sum == num;
    }

}

File: src/javascript/disariumNumber.js
function countDigits(num) {
    let len = 0;

    while(num > 0) {
        len++;
        num = Math.floor(num/10);
    }

    return len;
}

const isDisariumNumber = (num) => {
    let tempNum = num;
    let sum = 0;
    let digits = countDigits(num);

    while(tempNum > 0) {
        sum += Math.pow(tempNum%10, digits--);
        tempNum = Math.floor(tempNum/10);
    }

    return sum === num;
}

export default isDisariumNumber;
